session/handler.go

```diff
@@ -349,6 +366,83 @@ func (h *Handler) adminListSessions(w http.ResponseWriter, r *http.Request, ps h
 	h.r.Writer().Write(w, r, sess)
 }
 
+// Session Get Request
+//
+// The request object for getting a session in an administrative context.
+//
+// swagger:parameters adminGetSession
+// nolint:deadcode,unused
+type adminGetSessionRequest struct {
+	// ExpandOptions is a comma seperated list of all properties that must be expanded in the Session. If no value is provided, the expandable properties are skipped.
+	//
+	// enum: ["Identity","Devices"]
+	// required: false
+	// in: query
+	ExpandOptions string `json:"expand"`
+
+	// ID is the session's ID.
+	//
+	// required: true
+	// in: path
+	ID string `json:"id"`
+}
+
+// swagger:route GET /admin/sessions/{id} v0alpha2 adminGetSession
+//
+// This endpoint returns the session object with expandables specified.
+//
+// This endpoint is useful for:
+//
+// - Getting a session object with all specified expandables that exist in an administrative context.
+//
+//	Schemes: http, https
+//
+//	Security:
+//	  oryAccessToken:
+//
+//	Responses:
+//	  200: session
+//	  400: jsonError
+//	  401: jsonError
+//	  404: jsonError
+//	  500: jsonError
+func (h *Handler) adminGetSession(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	if ps.ByName("id") == "whoami" {
+		// for /admin/sessions/whoami redirect to the public route
+		x.RedirectToPublicRoute(h.r)(w, r, ps)
+		return
+	}
+
+	sID, err := uuid.FromString(ps.ByName("id"))
+	if err != nil {
+		h.r.Writer().WriteError(w, r, herodot.ErrBadRequest.WithError(err.Error()).WithDebug("could not parse UUID"))
+		return
+	}
+
+	var expandables Expandables
+
+	urlValues := r.URL.Query()
+	if urlValues.Has("expand") {
+		es := strings.Split(urlValues.Get("expand"), ",")
```

Comment:
An alternative would be to design the API like this:

```
?expand=foo&expand=bar&expand=baz
```

WDYT?

Reply:
@aeneasr I went through some reading for this design. (https://medium.com/raml-api/arrays-in-query-params-33189628fa68).

> What it looks like:
> 
> `?fields=kind,items(title,characteristics/length)`

The [Google drive approach](https://developers.google.com/drive/api/guides/performance#partial-response) enables us to have fine grained fields in addition to property expands (not currently supported). If this isn't something we need, then what you suggested seems better

Reply:
Cool, that makes sense! Can you please add this to the API design document once the work around the session listing is done? :)